File: gamebet_backend_python/shared/odds_format.py
def format_odds(value) -> float:
    try:
        n = float(value)
    except (TypeError, ValueError):
        return 0.0
    if not (n == n) or n == 0:  # isnan check
        return 0.0
    return round(n * 1000) / 1000


def format_bet_odds(bet: dict) -> dict:
    if not bet or not isinstance(bet, dict):
        return bet
    return {
        **bet,
        "HomeOdds": format_odds(bet.get("HomeOdds")),
        "AwayOdds": format_odds(bet.get("AwayOdds")),
    }

File: changmen/gamebet_backend/core/shared/odds_format.js
"use strict";

/** 赔率统一保留 3 位小数（与 TJ01 / A8 展示一致） */
function formatOdds(value) {
  const n = Number(value);
  if (!Number.isFinite(n)) return 0;
  if (n === 0) return 0;
  return Math.round(n * 1000) / 1000;
}

function formatBetOdds(bet) {
  if (!bet || typeof bet !== "object") return bet;
  return {
    ...bet,
    HomeOdds: formatOdds(bet.HomeOdds),
    AwayOdds: formatOdds(bet.AwayOdds),
  };
}

module.exports = {
  formatOdds,
  formatBetOdds,
};
